polygeist/tools/cgeist/Lib/TypeUtils.cc

```diff
@@ -451,4 +451,14 @@ unsigned getPrimitiveSizeInBits(mlir::Type Ty) {
           [](auto) -> unsigned { llvm_unreachable("Invalid primitive type"); });
 }
 
+/// There must be at least an argument and it must be a memref to a SYCL type.
+bool areSYCLMemberFunctionOrConstructorArgs(mlir::TypeRange Types) {
+  return !Types.empty() &&
+         TypeSwitch<mlir::Type, bool>(Types[0])
+             .Case<mlir::MemRefType, mlir::LLVM::LLVMPointerType>([](auto Ty) {
```

Comment:
have you ever seen llvm pointer type element being SYCL type? I thought we always generate memref of SYCL type.

Reply:
I honestly don't think so. Will remove this branch. Thanks!